app/gui/view/graph-editor/src/builtin/visualization/native/text_visualization/grid_view_entry.rs

```diff
@@ -18,10 +19,48 @@ use std::fmt::Write;
 // === Model ===
 // =============
 
-/// Model that contains the data that is required to populate the data in an `Entry`.
+/// Model that contains the data that is required to create and render an `Entry`.
 #[derive(Clone, Debug, Default)]
 pub struct Model {
-    pub text: String,
+    pub content:  Content,
+    pub bg_color: color::Rgba,
+}
+
+impl From<String> for Model {
+    fn from(content: String) -> Self {
+        let content = content.into();
+        let bg_color = color::Rgba::transparent();
+        Self { content, bg_color }
+    }
+}
+
+/// Content that is shown in an `Entry`.
+#[derive(Clone, Debug)]
+pub enum Content {
+    Text { content: String },
+    Divider { top: bool, bottom: bool, left: bool, right: bool },
+}
+
+impl Default for Content {
+    fn default() -> Self {
+        Self::Text { content: String::default() }
+    }
+}
+
+impl From<String> for Content {
+    fn from(content: String) -> Self {
+        Self::Text { content }
+    }
+}
+
+impl Model {
+    /// Get the content of the model.
+    pub fn text(&self, width: usize) -> String {
+        match &self.content {
+            Content::Text { content, .. } => content.to_string(),
+            _ => " ".repeat(width),
+        }
+    }
```

Comment:
What is `width` exactly? Why once we return string of given width, but in other case return content without adjusting width?

Reply:
Removed the parameter. This was used earlier for some formatting but is not needed any more. The current use is obsolete and can return an empty string instead.